homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -139,5 +135,5 @@ def extra_state_attributes(self) -> dict[str, Any]:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
-            self._state = state.state
+        if state := await self.async_get_last_sensor_data():
+            self._state = cast(float, state.native_value)
```

Comment:
Why do we need to cast?

Reply:
Euhm, honestly I just searched for this function and saw that they did it. Not sure

Reply:
Ah there is the reason
`Incompatible types in assignment (expression has type "str | int | float | date | datetime | Decimal | None", variable has type "str | int | float | None")  [assignment]`

Reply:
But it can be something else than float so probably cast it to `StateType` so it's same as the variable.

Reply:
Avoid the use of `StateType` as much as possible. More specific is better. Your comment/advise in this case, isn't an improvement.